test/libsolidity/SemanticTest.cpp

```diff
@@ -142,6 +147,31 @@ void SemanticTest::initializeBuiltins()
 		soltestAssert(_call.arguments.parameters.empty(), "No arguments expected.");
 		return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
 	};
+	m_hooks.emplace_back(
+		[](FunctionCall const& _call) -> std::vector<std::string>
+		{
+			if (_call.signature.find("effectsTest") != string::npos)
+				return {_call.signature + "0"};
+			return {};
+		}
+	);
+	m_hooks.emplace_back(
+		[](FunctionCall const& _call) -> std::vector<std::string>
+		{
+			if (_call.signature.find("effectsTest") != string::npos)
+				return {_call.signature + "1", _call.signature + "2"};
+			return {};
+		}
+	);
+}
+
+vector<string> SemanticTest::effectsOfCall(FunctionCall const& _call) const
+{
+	vector<string> effects;
+	for (auto const& hook: m_hooks)
+		for (auto const& effect: hook(_call))
+			effects.emplace_back(effect);
```

Comment:
The types are not long enough to warrant `auto` here. Also, we have a `+=` operator for `vector`s in `CommonData.h`:
```suggestion
	for (Hook const& hook: m_hooks)
		effects += hook(_call)
```